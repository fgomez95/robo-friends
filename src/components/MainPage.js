import React from 'react';
import Cards from './Cards';
import SearchBox from './SearchBox';

class MainPage extends React.Component {
  componentDidMount(){
    this.props.onGetRobots();
  }
  filterRobots = () => {
    return this.props.robots.filter(robot => {
      return robot.name.toLowerCase().includes(this.props.searchField);
    });
  }
  render(){
    const {
      searchField,
      onSearch
    } = this.props;
    
    return(
      <div className="tc">
        <h1 className="bg-black color-name white">
          Robo Friends
        </h1>
        <SearchBox 
          onChangeHandler={onSearch}
          value={searchField}
        />
        <Cards 
          data={this.filterRobots()}
        />
      </div>
    );
  }
}

export default MainPage;