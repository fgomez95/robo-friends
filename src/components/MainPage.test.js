import {shallow} from 'enzyme';
import React from 'react';
import MainPage from './MainPage';

let wrapper;
beforeEach(() => {
  const mockProps = {
    onGetRobots: jest.fn(),
    robots: [],
    searchField: '',
  }
  wrapper = shallow(<MainPage {...mockProps} />);
});

describe('Main Page', () => {
  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('filter the robots correctly', () => {
    const mockProps2 = {
      onGetRobots: jest.fn(),
      robots: [{
        id: 3,
        name: 'Fernando',
        email: 'fgomez@gmail.com',
      }],
      searchField: 'fuck you',
    }
    const wrapper2 = shallow(<MainPage {...mockProps2}/>);
    expect(wrapper.instance().filterRobots()).toEqual([]);
    expect(wrapper2.instance().filterRobots()).toEqual([]);
  });
  it('filter the robots correctly 2', () => {
    const mockProps3 = {
      onGetRobots: jest.fn(),
      robots: [{
        id: 3,
        name: 'Fernando',
        email: 'fgomez@gmail.com',
      }],
      searchField: 'fer',
    }
    const wrapper3 = shallow(<MainPage {...mockProps3}/>);
    expect(wrapper3.instance().filterRobots()).toEqual(mockProps3.robots);
  });
});