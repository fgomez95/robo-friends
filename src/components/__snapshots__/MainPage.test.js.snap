// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Page renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainPage
    onGetRobots={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    robots={Array []}
    searchField=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="bg-black color-name white"
        >
          Robo Friends
        </h1>,
        <searchBox
          value=""
        />,
        <Cards
          data={[Function]}
        />,
      ],
      "className": "tc",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Robo Friends",
          "className": "bg-black color-name white",
        },
        "ref": null,
        "rendered": "Robo Friends",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onChangeHandler": undefined,
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="bg-black color-name white"
          >
            Robo Friends
          </h1>,
          <searchBox
            value=""
          />,
          <Cards
            data={[Function]}
          />,
        ],
        "className": "tc",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Robo Friends",
            "className": "bg-black color-name white",
          },
          "ref": null,
          "rendered": "Robo Friends",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onChangeHandler": undefined,
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "data": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
